lib/gvl-tracing.rb

```diff
@@ -28,6 +28,53 @@
 require_relative "gvl_tracing/version"
 
 require "gvl_tracing_native_extension"
+require "json"
 
 module GvlTracing
+  @path = "/tmp/gvl-tracing.json"
+
+  class << self
+    private :_start
+    private :_stop
+
+    def start(file)
+      @path = file
+      _start(@path)
+    end
+
+    def stop
+      thread_list = Thread.list
+
+      _stop
+
+      append_thread_name(thread_list)
+    end
+
+    private
+
+    def append_thread_name(list)
+      threads_name = aggreate_thread_list(list).join(",\n")
+      File.open(@path, 'a') do |f|
+        f.puts(threads_name)
+        f.puts("]")
+      end
+    end
+
+    def aggreate_thread_list(list)
+      list.each_with_object([]) do |t, acc|
+        next unless t.name
+
+        acc << {"ph": "M", "pid": Process.pid, "tid": t.native_thread_id, "name": "thread_name", "args": {"name": thread_label(t)}}.to_json
```

Comment:
Minor: To be honest, this line is so close to the final output JSON, that I'm not sure it's worth using `to_json` ;)